### PR TITLE
PRIME-1551 Integration testing with PLR

### PR DESCRIPTION
**Pull request by neophyte57**
Completed initial end-to-end integration testing with Gary (PLR), with mutual authentication occurring.  

White-listing PLR IP addresses not to be implemented

Can be tested locally like this:  `curl --insecure --header "@../Utils/Auth/client-pem-url-encoded-headers.txt" --data-binary "@PRPM_IN303030CA.xml" https://127.0.0.1:5001/api/PLRHL7`  
image

or remotely like this `curl --insecure --cert client.crt:changeme --key client.key --data-binary "@PRPM_IN303030CA.xml" https://mutualauth-pr-1247.pharmanetenrolment.gov.bc.ca/api/v1/PLRHL7`

### CODE REVIEW

File: prime-dotnet-webapi/Services/SoapService.cs [hunk 1]
```diff
@@ -151,9 +166,20 @@ private PlrProvider ReadDistributionMessage(string messageContent)
...
             }
             result.Suffix = ReadNodeData($"//{Prefix}:healthCarePrincipalPerson/{Prefix}:name[@use='L']/{Prefix}:suffix", documentRoot, messageId);
-            string[] telephoneNumberParts = SplitHL7v3TelecomNumber(RemoveHL7v3TelecomType(ReadNodeData($"//{Prefix}:healthCareProvider/{Prefix}:telecom[@use='WP' and starts-with(@value, 'tel')]/@value", documentRoot, messageId)));
-            result.TelephoneAreaCode = telephoneNumberParts[0];
-            result.TelephoneNumber = telephoneNumberParts[1];
+            string telephoneNumData = ReadNodeData($"//{Prefix}:healthCareProvider/{Prefix}:telecom[@use='WP' and starts-with(@value, 'tel')]/@value", documentRoot, messageId);
+            if (telephoneNumData != null)
+            {
+                string[] telephoneNumberParts = SplitHL7v3TelecomNumber(RemoveHL7v3TelecomType(telephoneNumData));
+                if (telephoneNumberParts.Length == 2)
```
Comment (johnnyl6) [inline prime-dotnet-webapi/Services/SoapService.cs:173]: Is it supposed to be `faxNumberParts.Length == 2` ?

File: prime-dotnet-webapi/Services/SoapService.cs [hunk 2]
```diff
@@ -134,12 +134,27 @@ private PlrProvider ReadDistributionMessage(string messageContent)
...
             result.Expertise = ReadMultiNodeData($"//{Prefix}:healthCareProvider/{Prefix}:relatedTo/{Prefix}:qualifiedEntity/{Prefix}:code[not({Prefix}:originalText/{Prefix}:reference)]/@code", documentRoot, messageId);
-            string[] faxNumberParts = SplitHL7v3TelecomNumber(RemoveHL7v3TelecomType(ReadNodeData($"//{Prefix}:healthCareProvider/{Prefix}:telecom[@use='WP' and starts-with(@value, 'fax')]/@value", documentRoot, messageId)));
-            result.FaxAreaCode = faxNumberParts[0];
-            result.FaxNumber = faxNumberParts[1];
+            string faxNumberData = ReadNodeData($"//{Prefix}:healthCareProvider/{Prefix}:telecom[@use='WP' and starts-with(@value, 'fax')]/@value", documentRoot, messageId);
+            if (faxNumberData != null)
+            {
+                string[] faxNumberParts = SplitHL7v3TelecomNumber(RemoveHL7v3TelecomType(faxNumberData));
+                if (faxNumberData.Length == 2)
+                {
```
Comment (johnnyl6) [inline prime-dotnet-webapi/Services/SoapService.cs:149]: Is it supposed to be `faxNumberParts.Length == 2`?

File: prime-dotnet-webapi/Services/SoapService.cs [hunk 3]
```diff
@@ -177,7 +203,8 @@ public static string[] SplitHL7v3TelecomNumber(string telecomNumber)
             }
             else
             {
-                throw new ArgumentOutOfRangeException(nameof(telecomNumber), telecomNumber, "Not a 10 digit string");
+                return new string[] { telecomNumber };
+                // throw new ArgumentOutOfRangeException(nameof(telecomNumber), telecomNumber, "Not a 10 digit string");
```
Comment (johnnyl6) [inline prime-dotnet-webapi/Services/SoapService.cs:207]: Seems safe to remove the comment

**Comment by neophyte57** [comment]
Good catch @johnnyl6 .  I addressed your comments and also a warning in the back-end log "The property 'Credentials' on entity type 'PlrProvider' is a collection or enumeration type with a value converter but with no value comparer. Set a value comparer to ensure the collection/enumeration elements are compared correctly."

**Review by johnnyl6 — APPROVED** [review]
LGTM

**Review by mtpultz — APPROVED** [review]
image

**Pull request merged**